src/components/routes/EventPage.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react';
+import { withRouter } from 'react-router-dom';
+import { Button, Container, Divider, Grid, LinearProgress, Paper, Typography } from '@material-ui/core';
+import * as stockImage from '../../res/stockEventPhoto.jpg';
+import AccessTimeIcon from '@material-ui/icons/AccessTime';
+import useStyles from '../util/useStyles';
+
+function EventPage() {
+	const [loading, setLoading] = useState<boolean>(false);
+	const [name, setName] = useState<string>('');
+	const [startDate, setStartDate] = useState<Date | string>('');
+	const [endDate, setEndDate] = useState<Date | string>('');
+	const [description, setDescription] = useState<string>('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut\n' +
+        '                            labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco\n' +
+        '                            laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in\n' +
+        '                            voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat\n' +
+        '                            cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laboruLorem\n' +
+        '                            ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut\n' +
+        '                            labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco\n' +
+        '                            laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in\n' +
+        '                            voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat\n' +
+        '                            cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laboru');
+	const [image, setImage] = useState(stockImage);
+	const [channelID, setChannelID] = useState<string>('ggg');
+	const classes = useStyles();
+
+	useEffect(() => {
+		// expecting API CALL
+		setLoading(true);
+		setName('Hello');
+		setStartDate('2/12');
+		setEndDate('23/12');
+		// setDescription('An Amazing event');
+		setChannelID('eeee');
+		setLoading(false);
+	}, []);
```

Comment:
Could you use a single state object for this? This would be quite a lot of calls to the state and would lead to multiple rerenders unnecessarily.

Also, does this even need to be a stateful component? I don't see event data changing that frequently, this makes more sense as props imo. What do you think @HishamYahya?

Reply:
im not sure if we want it to be stateful or not because im not sure if you get all the data required from the start

Reply:
Ok, can you still make this a single object for the state though?